### ISSUE TITLE
setInterval RangeError: Maximum call stack

### ISSUE THREAD

**Issue by johnlindquist** [angular/zone.js#57]
Using the preview here: https://github.com/angular/quickstart

Adding `setInterval` anywhere throws an error

```
es6-shim.js:29889 Uncaught RangeError: Maximum call stack size exceeded
es6-shim.js:29889 run
es6-shim.js:29866 zoneBoundFn
es6-shim.js:29940 fnNames.map.forEach.zone.(anonymous function).arguments.(anonymous function)
es6-shim.js:29873 (anonymous function)
es6-shim.js:29889 runes6-shim.js:29866 zoneBoundFn
es6-shim.js:29940 fnNames.map.forEach.zone.(anonymous function).arguments.(anonymous function)
es6-shim.js:29873 (anonymous function)
es6-shim.js:29889 run
es6-shim.js:29866 zoneBoundFn
```

**Comment by Invis1ble** [comment]
The same issue:
```
RangeError: Maximum call stack size exceeded
STACKTRACE:
RangeError: Maximum call stack size exceeded
    at join (native)
    at VmTurnZone.$__export._onError (http://angular/quickstart/angular2/src/core/zone/vm_turn_zone.js:84:44)
    at Zone.errorHandling.onError (http://angular/quickstart/angular2/src/core/zone/vm_turn_zone.js:58:30)
    at Zone.run (http://angular/quickstart/dist/es6-shim.js:29892:14)
    at zoneBoundFn (http://angular/quickstart/dist/es6-shim.js:29866:19)
    at arguments.(anonymous function) (http://angular/quickstart/dist/es6-shim.js:29928:23)
    at Zone.run (http://angular/quickstart/dist/es6-shim.js:29889:19)
    at zoneBoundFn (http://angular/quickstart/dist/es6-shim.js:29866:19)
    at arguments.(anonymous function) (http://angular/quickstart/dist/es6-shim.js:29928:23)
    at Zone.run (http://angular/quickstart/dist/es6-shim.js:29889:19) ```

**Comment by pkozlowski-opensource** [comment]
Same with `setTimeout` - interestingly enough, this only happens when zones are included _before_ ng2 - when you push them after ng2 things seem to work as expected....

**Comment by pkozlowski-opensource** [comment]
Actually, after some more testing, it doesn't seem to be linked to scripts inclusion order.

**Comment by rkirov** [comment]
The issue is in the following line - https://github.com/angular/zone.js/blob/3a3dae4295657fb8e09f6025cabb984fed87a38f/zone.js#L129

In non-strict mode, modifying `arguments[0]` is actually reflected in the variable used as argument 0. see: http://plnkr.co/edit/IqsEew9ihO0oC2itrLo9

**Comment by vicb** [comment]
closed via #70
